### ISSUE TITLE
DeprecationWarning: `context.fix` is being deprecated.

### ISSUE THREAD

**Issue by elanora96** [csstools/stylelint-prefer-nesting#20]
<details>

<summary>

# My machine and project (details)

</summary>

## Machine, node, and npm
```sh
❯ uname -ropm && node --version && npm --version
6.10.4-arch2-1 x86_64 unknown GNU/Linux
v22.6.0
10.8.2
```
## `package.json` snippet
```json
  "dependencies": {
    "dayjs": "^1.11.12",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@eslint/js": "^9.9.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@vitejs/plugin-react-swc": "^3.5.0",
    "eslint": "^9.9.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-prettier": "^5.2.1",
    "eslint-plugin-react": "^7.35.0",
    "eslint-plugin-react-hooks": "^5.1.0-rc.0",
    "eslint-plugin-react-refresh": "^0.4.9",
    "globals": "^15.9.0",
    "husky": "^9.1.4",
    "lint-staged": "^15.2.9",
    "prettier": "^3.3.3",
    "stylelint": "^16.8.2",
    "stylelint-config-clean-order": "^6.1.0",
    "stylelint-config-standard": "^36.0.1",
    "stylelint-no-indistinguishable-colors": "^2.3.0",
    "stylelint-use-nesting": "^5.1.2",
    "typescript": "^5.5.3",
    "typescript-eslint": "^8.0.1",
    "vite": "^5.4.1",
    "vite-tsconfig-paths": "^5.0.1"
  }
```

## `stylelint.config.js`
```ts
/**
 * @type {import('stylelint').Config}
 */
export default {
  ignoreFiles: ['node_modules/**/*', 'dist/**/*'],
  extends: ['stylelint-config-standard', 'stylelint-config-clean-order'],
  plugins: ['stylelint-no-indistinguishable-colors', 'stylelint-use-nesting'],
  rules: {
    'plugin/stylelint-no-indistinguishable-colors': true,
    'csstools/use-nesting': true,
    'selector-class-pattern': '^[a-z][a-zA-Z0-9]+$|^[A-Z][a-zA-Z0-9]+$',
    'keyframes-name-pattern': '^[a-z][a-zA-Z0-9]+$|^[A-Z][a-zA-Z0-9]+$',
  },
};
```

</details>

# Problem

Running stylelint with this package enabled produces many deprecation warnings referencing `content.fix`

```sh
❯ npx stylelint "**/*.css" NODE_OPTIONS=--trace-deprecation
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
(node:189960) [stylelint:005] DeprecationWarning: `context.fix` is being deprecated.
Please pass a `fix` callback to the `report` utility of "csstools/use-nesting" instead.
```

I haven't personally had a chance to truly investigate the warning yet. While this doesn't *appear* to affect my usability, I would like to assist in tracking down the cause of this.

**Comment by Mouvedia** [comment]
Normally you should get one deprecation warning per rule.
This rule is unusual in that it accesses it 7 times.
Maybe it's related to the usage of

https://github.com/csstools/stylelint-use-nesting/blob/27503ef65ab990d324972142d48bd82da0ebaa85/index.mjs#L19

Until the plugin is refactored to use the `fix` callback you can either use [`--no-deprecation`](https://nodejs.org/api/cli.html#--no-deprecation) or [`--quiet-deprecation-warnings`](https://stylelint.io/user-guide/cli#--quiet-deprecation-warnings).

**Comment by romainmenke** [comment]
Thank you @elanora96 for reaching out about this.

This was fixed in `6.0.0` : https://github.com/csstools/stylelint-use-nesting/blob/main/CHANGELOG.md#600

Please let us know if you encounter further issues.
